rules/numeric-host/src/lib.rs

```diff
@@ -0,0 +1,138 @@
+use anyhow::{anyhow, Ok};
+use bulwark_wasm_sdk::*;
+use regex::*;
+
+struct NumericHostPlugin;
+
+fn check_numeric_host(req: Request) -> std::result::Result<bool, anyhow::Error> {
+    let host = req
+        .headers()
+        .get("Host")
+        .ok_or(anyhow!("Missing Host Header."))?;
+
+    let re_ipv4 = Regex::new(r"^((25[0-5]|(2[0-4]|1\d|[1-9]|)\d)\.?\b){4}(:[0-9]{1,5})?$").unwrap();
+    let re_ipv6 = Regex::new(r"\[(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}\](:[0-9]{1,5})?|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))").unwrap();
+    // numerals as IPv4
+    let re_decimal = Regex::new(r"^[0-9]{1,10}(:[0-9]{1,5})?$").unwrap();
+    let re_binary = Regex::new(r"^0[bB][0-1]{1,32}(:[0-9]{1,5})?$").unwrap();
+    let re_hex = Regex::new(r"^0[xX][0-9a-fA-F]{1,8}(:[0-9]{1,5})?$").unwrap();
+    let re_octal = Regex::new(r"^0[0-7]{1,11}(:[0-9]{1,5})?$").unwrap();
+
+    let match_ipv4 = re_ipv4.find(host.to_str()?);
+    let match_ipv6 = re_ipv6.find(host.to_str()?);
+    let match_decimal = re_decimal.find(host.to_str()?);
+    let match_binary = re_binary.find(host.to_str()?);
+    let match_hex = re_hex.find(host.to_str()?);
+    let match_octal = re_octal.find(host.to_str()?);
+
+    match (
+        match_ipv4,
+        match_ipv6,
+        match_decimal,
+        match_binary,
+        match_hex,
+        match_octal,
+    ) {
+        (None, None, None, None, None, None) => Ok(false),
+        _ => Ok(true),
+    }
+}
+
+#[bulwark_plugin]
+impl Handlers for NumericHostPlugin {
+    fn on_request_decision() -> Result {
+        if check_numeric_host(get_request())? {
+            set_restricted(1.0);
+        }
+        Ok(())
+    }
+
+    fn on_request_body_decision() -> Result {
```

Comment:
I think the second handler might not be needed since the body doesn't affect the outcome of the detection, but I'm not 100% certain on that. Might be worth double-checking?